CI build for SDE

TSAnsari opened the pull request.

TSAnsari committed:
CI build for PELUX Software Development Environment

Build will be triggered on nodes with labels,

PyTest & Vagrand & VirtualBox

Signed-off-by: tansari <tansari@luxoft.com>

gunnarx reviewed (commented):
The commit summary should also change CDE->SDE, but it's still WIP so I'm sure it will...

gunnarx commented on test/test_sdk_installation.py:
```diff
@@ -53,6 +53,9 @@ def pipe_to_str(pipe):
 @pytest.fixture(scope="class")
 def vagrant():
     try:
+        #print(os.environ['HOME'])
```
Just note that it will conflict with the deduplication in #13 (which is based on current master).  We'll see what gets merged first and the other PR will need a refresh.

TSAnsari replied:
This is just some experimentation stuff. This will not be upstreamed. This can be ignored.

GordanM commented on Jenkinsfile:
```diff
@@ -10,11 +10,9 @@
+node ("PyTest && Vagrant && VirtualBox") {
     stage("Checkout") {
         checkout scm
-        sh 'git submodule update --init'
```
Is there a reason to remove this line? :)

TSAnsari replied:
There are no submodules required by this repo

GordanM replied:
Yup that is true, but it doesn't hurt does it, can you comment @thenor ? I mean it creates a bigger diff against other jenkinsfiles and we might add a submodule at some point :). 

Furthermore, even if we include this change it should be a separate commit due to the fact that changes need to be atomic.

TSAnsari committed:
dlt-viewer: fixed dlt-viewer installation test

Fixed the following issues to pass the test,

- It seems someone forced commit on the dlt-viewer master resulting
  in changing the commit hash. So the test could not find the hash
  and fails. This commit changes the revision to the latest commit
  hash.

- The vagrant setup of the VirtualBox did not installed Qt serialport
  module, resulting in the installation of dlt-viewer to fail. This
  commit adds the installation of Qt serialport module.

Signed-off-by: tansari <tansari@luxoft.com>

TSAnsari commented:
I have made a couple of changes. Please review.

GordanM reviewed (approved):
LGTM
